src/common/dav/tx.c

```diff
@@ -507,7 +507,25 @@ dav_tx_begin(dav_obj_t *pop, jmp_buf env, ...)
 
 		VALGRIND_START_TX;
 	} else if (tx->stage == DAV_TX_STAGE_NONE) {
+		struct umem_wal_tx *utx;
+
 		DAV_DBG("");
+		utx = pop->do_utx;
+		if (utx == NULL) {
+			utx = dav_umem_wtx_new(pop);
+			if (utx == NULL) {
+				err = ENOMEM;
+				goto err_abort;
+			}
+		}
+		err = dav_wal_tx_reserve(pop);
+		if (err) {
+			D_ERROR("so_wal_reserv failed, "DF_RC"\n", DP_RC(err));
+			D_FREE(utx);
+			pop->do_utx = NULL;
+			goto err_abort;
+		}
+		tx = get_tx();
```

Comment:
get_tx() returns per-thread tx structure pointer, it is the same for all ULTs on this pthread. It is not a ULT local store data. Re-loading the same tx address doesn't serve the purpose of UTL's data separation.

Reply:
Yes, this call is redundant. Is it ok I clean it up if future PR?